src/main/java/org/cryptomator/integrations/update/UpdateService.java

```diff
@@ -0,0 +1,119 @@
+package org.cryptomator.integrations.update;
+
+import org.cryptomator.integrations.common.DistributionChannel;
+import org.cryptomator.integrations.common.IntegrationsLoader;
+
+import java.util.stream.Stream;
+
+/**
+ * This is the interface used by Cryptomator to provide a way to update Cryptomator in a convinient way.
+ * It's idependent of the supported platforms and package distribution channels.
+ */
+public interface UpdateService {
+
+	static Stream<UpdateService> get() {
+		return IntegrationsLoader.loadAll(UpdateService.class).filter(UpdateService::isSupported);
+	}
+
+	/**
+	 * @return <code>true</code> if this UppdateService can update the app.
+	 * @implSpec This method must not throw any exceptions and should fail fast
+	 * returning <code>false</code> if it's not possible to use this UppdateService
+	 */
+	boolean isSupported();
+
+	/**
+	 * Retrieve an object to check for the latest release published on the given channel.
+	 *
+	 * @param channel The DistributionChannel.Value to check.
+	 * @return        An object that is capable of checking asynchronously for the latest release.
+	 */
+	Object getLatestReleaseChecker(DistributionChannel.Value channel);
```

Comment:
Checking whether an update is available can be a simple blocking call. KISS

Reply:
Understood, will do so.

Reply:
I meant a simple blocking `boolean isUpdatePresent()` method - optionally with a build number or the like.

What is important: the API user should only use the API. It must not know about the implementation. I.e. these kinds of casts in the main app are disallowed:

```java
startCheckingWithFlatpakUpdater((UpdateCheckerTask) updateChecker.getUpdater(env.getBuildNumber()), Duration.ZERO)
```

Reply:
The point is: The main app shall *not know* what update mechanism is used. "code against interfaces". The word "Flatpak" must not appear in downstream code (except for the display name).